src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkUtils.java

```diff
@@ -611,6 +612,13 @@ private static BigQueryFileFormat getFileFormat(List<BigQueryTableFieldSchema> f
       if (DATETIME.equals(field.getType())) {
         return BigQueryFileFormat.NEWLINE_DELIMITED_JSON;
       }
+      // If the field name is not in english characters, then we will use json format
+      // We do this as the avro load job in BQ does not support non-english characters in field names for now
+      String fieldName = field.getName();
+      String wordRegex = "[\\w]+";
```

Comment:
does this cover all special characters like underscore or digits that were supported previously by BQ. If not use the same REGEX that was given in bigqueryConnector library.

Reply:
This does not cover everything this only checks for [a-z A-Z 0-9],
we are doing a negative here, so it means as long as we don't have a character outside of  [a-z A-Z 0-9] we can use avro format, else we use the JSON format.

Reply:
in case of underscore also, it will write into json then, which is not required